src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -621,27 +436,36 @@ public string GetTraceDataToSave()
             if (!this.traceData.Any(srtd => srtd.HasAnyNestedData))
                 return null;
 
-            TraceSerialiserHelper helper = new TraceSerialiserHelper();
-            helper.TraceData = this.traceData;
-
-            using (MemoryStream memoryStream = new MemoryStream())
-            {
+            var serializedTraceData = JsonConvert.SerializeObject(this.traceData);
+            //Optional
+            serializedTraceData= CompressSerializedTraceData(serializedTraceData);
+            return serializedTraceData;
+        }
 
-                IFormatter formatter = new SoapFormatter();
-                formatter.Serialize(memoryStream, helper);
+        /// <summary>
+        /// Compress the input string via GZIP to Base64String
+        /// </summary>
+        /// <param name="json"></param>
+        /// <returns></returns>
+        public static string CompressSerializedTraceData(string json)
```

Comment:
do you anticipate host integrators needing this method?

Reply:
Probably not